File: problem-solving/baekjoon/2309_일곱_난쟁이.py
# https://www.acmicpc.net/problem/2309
import sys

if __name__ == "__main__":
    heights = []
    total = 0
    for i in range(0, 9):
        h = int(sys.stdin.readline())
        heights.append(h)
        total += h

    heights.sort()
    stop = False
    for i in range(0, 8):
        for j in range(i+1, 9):
            if heights[i]+heights[j] == total-100:
                for k in range(0, 9):
                    if k != i and k != j:
                        print(heights[k])
                stop = True
                break
        if stop:
            break

File: problem-solving/baekjoon/2309_일곱_난쟁이.go
// https://www.acmicpc.net/problem/2309
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	heights := []int{}
	var sum int
	for i := 0; i < 9; i++ {
		var h int
		fmt.Fscanln(reader, &h)
		heights = append(heights, h)
		sum += h
	}

	sort.Ints(heights)
	for i := 0; i < 8; i++ {
		for j := i + 1; j < 9; j++ {
			if heights[i]+heights[j] == sum-100 {
				for k := 0; k < 9; k++ {
					if k != i && k != j {
						fmt.Fprintln(writer, heights[k])
					}
				}
				return
			}
		}
	}
}
